src/globals.c

```diff
@@ -69,6 +74,121 @@ char *elf_symtab;
 char *elf_strtab;
 char *elf_section;
 
+/**
+ * arena_block_create() - creates a new arena block with given capacity.
+ * The created arena block is guaranteed to be zero-initialized.
+ * @capacity: The capacity of the arena block. Must be positive.
+ *
+ * Return: The pointer of created arena block. NULL if failed to allocate.
+ */
+arena_block_t *arena_block_create(int capacity)
+{
+    arena_block_t *block = malloc(sizeof(arena_block_t));
+
+    if (!block) {
+        printf("Failed to allocate memory for arena block\n");
+        exit(1);
+    }
+
+    block->memory = calloc(capacity, sizeof(char));
+
+    if (!block->memory) {
+        printf("Failed to allocate memory for arena block\n");
+        free(block);
+        exit(1);
+    }
+
+    block->capacity = capacity;
+    block->offset = 0;
+    block->next = NULL;
+    return block;
+}
+
+/**
+ * arena_init() - initializes the given arena with initial capacity.
+ * @initial_capacity: The initial capacity of the arena. Must be positive.
+ *
+ * Return: The pointer of initialized arena.
+ */
+arena_t *arena_init(int initial_capacity)
+{
+    arena_t *arena = malloc(sizeof(arena_t));
+    arena->head = arena_block_create(initial_capacity);
+    return arena;
+}
+
+/**
+ * arena_alloc() - allocates memory from the given arena with given size.
+ * The arena may create a new arena block if no space is available.
+ * @arena: The arena to allocate memory from. Must not be NULL.
+ * @size: The size of memory to allocate. Must be positive.
+ *
+ * Return: The pointer of allocated memory. NULL if new arena block is failed to
+ * allocate.
+ */
+void *arena_alloc(arena_t *arena, int size)
+{
+    char *ptr;
+    arena_block_t *block = arena->head;
+
+    while (block) {
+        if (block->offset + size <= block->capacity) {
+            ptr = block->memory + block->offset;
+            block->offset += size;
+            return ptr;
+        }
+        if (!block->next)
+            break;
+        block = block->next;
+    }
+
+    /* If no space is available, create a new block
+     * Allocate at least 1 KB or the requested size
+     */
```

Comment:
Is the description inconsistent since the default arena size is defined as 256KiB block at `#define DEFAULT_ARENA_SIZE 262144`?